### PR TITLE
Refs task #368940 List users

### PR DESCRIPTION
**Pull request by nguyenthinhhn**
## Related Tickets

- [#368940](https://dev.sun-asterisk.com/issues/368940)

## What's this PR do ?

- [x] Implement function List users

## Library
- Library: yajra/laravel-datatables-oracle": "^9.7"

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [ ] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>failure</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        return view('admins/user');
+    }
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 22</br>Opening brace should be on a new line

Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.FunctionOpeningBraceSpace.SpacingAfter</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 22</br>Expected 0 blank lines after opening function brace; 1 found

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        return Datatables::of(collect(UserResource::collection(User::all())))->toJson();
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionClosingBrace.SpacingBeforeClose</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 35</br>Function closing brace must go on the next line following the body; found 1 blank lines before brace

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+use App\Models\User;
+use App\Http\Resources\User as UserResource;
+
+class UserController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.FunctionOpeningBraceSpace.SpacingAfter</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 18</br>Expected 0 blank lines after opening function brace; 1 found

File: app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    public function index()
+    {
+
+        return view('admins/user');
+    }
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 23</br>Opening brace should be on a new line

Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.FunctionOpeningBraceSpace.SpacingAfter</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 23</br>Expected 0 blank lines after opening function brace; 1 found

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/UserController.php [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+use Yajra\Datatables\Datatables;
+use App\Models\User;
+use App\Http\Resources\User as UserResource;
+
+class UserController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     *
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 17</br>Opening brace should be on a new line

Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.FunctionOpeningBraceSpace.SpacingAfter</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 17</br>Expected 0 blank lines after opening function brace; 1 found

File: app/Http/Controllers/UserController.php [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+     */
+    public function index() {
+
+        return view('admins/user');
+    }
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 22</br>Opening brace should be on a new line

Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.FunctionOpeningBraceSpace.SpacingAfter</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 22</br>Expected 0 blank lines after opening function brace; 1 found

**Comment by framgiaciv3** [comment]
No data coverage!

File: app/Http/Controllers/UserController.php [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Yajra\Datatables\Datatables;
+use App\Models\User;
+use App\Http\Resources\User as UserResource;
+
+class UserController extends Controller
+{
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 13</br>Opening brace should be on a new line

Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.FunctionOpeningBraceSpace.SpacingAfter</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 13</br>Expected 0 blank lines after opening function brace; 1 found

File: app/Http/Controllers/UserController.php [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    
+    public function index() {
+
+        return view('admins/user');
+    }
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 18</br>Opening brace should be on a new line

Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.FunctionOpeningBraceSpace.SpacingAfter</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 18</br>Expected 0 blank lines after opening function brace; 1 found

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr></table>

File: routes/web.php [hunk 9]
```diff
@@ -12,15 +12,18 @@
...
 Auth::routes([
-  'register' => false,
-  'verify' => false,
+	'register' => false,
+	'verify' => false,
 ]);
 
 Route::namespace('Auth')->group(function () {
-    Route::get('/redirect', 'SocialAuthGoogleController@redirect');
-    Route::get('/callback', 'SocialAuthGoogleController@callback');
```
Comment (ghost) [inline routes/web.php:25]: bên php là indent tab 4 spaces phải không e

File: resources/views/layout/app.blade.php [hunk 10]
```diff
@@ -38,6 +38,7 @@
     <script src="{{ asset('js/app.js') }}"></script>
```
Comment (phucqd-1305) [inline resources/views/layout/app.blade.php]: mix(file_path) cho cái này

File: routes/web.php (hunk 9 above)
Comment (nguyenthinhhn) [reply routes/web.php:25]: đúng r a . 4 spaces đấy

File: resources/views/layout/app.blade.php (hunk 10 above)
Comment (nguyenthinhhn) [reply resources/views/layout/app.blade.php]: em mix thì cứ bị lỗi nên em để tạm vậy ạ

**Comment by ghost** [comment]
> ## Related Tickets
> * [#368940](https://dev.sun-asterisk.com/issues/368940)
> 
> ## What's this PR do ?
> * [x]  Implement function A
> 
> ## Library
> _(List package, library third party add new include version)_
> 
> * Library: google-analysis.js
> 
> ## Performance
> * [ ]  Resolved n + 1 query
> * [ ]  Run explain query already
> * [ ]  Time run rake task : 1000 ms
> 
> ## Checklist
> * [ ]  It was tested in local success?
> * [] Fill link PR into ticket and the opposite
> * [ ]  Note purpose/reason, solution, scope of influence into ticket
> 
> ## ENV note
> ## Notes
> _(Other notes)_

update lại cho đầy đủ thông tin e

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr></table>

File: resources/views/admins/user.blade.php [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+<div class="block full">
+    <div class="block-title">
+        <h2><strong>List user</strong></h2>
+    </div>
+
+    <div class="table-responsive">
+        <table id="user-table" class="table table-vcenter table-condensed table-bordered">
+            <thead>
+                <tr>
+                    <th class="text-center">ID</th>
```
Comment (ghost) [inline resources/views/admins/user.blade.php]: trong list không hiển thị ID đâu e

Comment (nguyenthinhhn) [reply resources/views/admins/user.blade.php]: ok a. e sửa thành STT rồi

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr></table>

**Review by ghost — APPROVED** [review]

File: app/Http/Controllers/UserController.php [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+{
+
+    public function index()
+    {
+        return view('admins/user');
+    }
+
+    public function getList()
+    {
+        return Datatables::of(collect(UserResource::collection(User::all())))->addIndexColumn()->make(true);
```
Comment (tamtt-1031) [inline app/Http/Controllers/UserController.php]: có order theo spec chưa em nhỉ?

File: public/js/user.js [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+                    render: function (data, type, row) {
+                        return '<div class="text-center"><a href="#" data-toggle="tooltip" title="Edit" class="btn btn-xs btn-default" onclick="edit(' + data + ');"><i class="fa fa-pencil"></i></a>' +
+                        '&nbsp<a href="#" data-toggle="tooltip" title="Delete" class="btn btn-xs btn-danger" onclick="delete(' + data + ');"><i class="fa fa-times"></i></a></div>';
+                    }
+                },
+                ]
+            });
+
+            /* Add placeholder attribute to the search input */
+            $('.dataTables_filter input').attr('placeholder', 'Search');
```
Comment (tamtt-1031) [inline public/js/user.js:68]: phần search sẽ khác spec khi mình dùng database table e xem có phải update lại spec tưowng tự như datatable hiện tại dùng để sau có QA họ cũng tiện viết testcase luôn nhé

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/UserController.php [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+
+class UserController extends Controller
+{
+
+    public function index()
+    {
+        return view('admins/user');
+    }
+
+    public function getList()
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Files.LineLength.TooLong</br>**Severity**: warning</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 20</br>Line exceeds 120 characters; contains 140 characters

Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Formatting.DisallowMultipleStatements.SameLine</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 20</br>Each PHP statement must be on a line by itself

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/UserController.php [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+{
+
+    public function index()
+    {
+        return view('admins/user');
+    }
+
+    public function getList(Request $request)
+    {
+        return Datatables::of(collect(UserResource::collection(User::orderBy('created_at', 'desc')->get())))
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Formatting.DisallowMultipleStatements.SameLine</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 22</br>Each PHP statement must be on a line by itself

**Comment by framgiaciv3** [comment]
No data coverage!

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr></table>

**Review by quanghung97 — APPROVED** [review]
OK

**Review by tamtt-1031 — APPROVED** [review]

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr></table>

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**